### PR TITLE
fix: strings

### PR DESCRIPTION
**Pull request by evilebottnawi**
All buggy things are commented.

Feel free to feedback.

Issue about syntax: https://github.com/prettier/plugin-php/issues/138.

Let's review this and merge as is. It is maximum how we can handle strings right now.

### CODE REVIEW

File: tests/string/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+// $string = "This is the value of the var named by the return value of \\$object->getName(): {\${$object->getName()}}";
+$string = "I'd like an {\${beers::softdrink}}\\n";
+$string = "I'd like an {\${beers::$ale}}\\n";
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<?php
+$string = 'this is a simple string';
+// $string = 'You can also have embedded\\n newlines in
+// strings this way as it is
+// okay to do';
+$string = 'Arnold once said: "I\\'ll be back"';
```
Comment (mgrip) [inline tests/string/__snapshots__/jsfmt.spec.js.snap]: Same thing - i think an extra `\` got added

File: tests/string/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+$string = "This is the value of the var named $name: {\${$name}}";
+$string = "This is the value of the var named by the return value of getName(): {\${getName()}}";
+$string = "This is the value of the var named by the return value of getName(): {\${getName($arg1, $arg2, $arg3)}}";
+// $string = "This is the value of the var named by the return value of \\$object->getName(): {\${$object->getName()}}";
+$string = "I'd like an {\${beers::softdrink}}\\n";
+$string = "I'd like an {\${beers::$ale}}\\n";
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<?php
+$string = 'this is a simple string';
+// $string = 'You can also have embedded\\n newlines in
```
Comment (mgrip) [inline tests/string/__snapshots__/jsfmt.spec.js.snap]: this should remain `'You can also have embedded\n newlines in`, right?

File: tests/string/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (mgrip) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: maybe add a test `$string = 'This is an escaped quote \' test';` - I'm curious if that works

File: tests/string/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (evilebottnawi) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: @mgrip yes

File: tests/string/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (evilebottnawi) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: @mgrip yep let's add test, but here all fine with `\`, jest always escaped special characters

Comment (mgrip) [reply tests/string/__snapshots__/jsfmt.spec.js.snap]: just realized that - thanks! looks good to me!

**Pull request merged**